### ISSUE TITLE
QEfficient.cloud.export not using the hf_token passed via cli flag

### ISSUE THREAD

**Issue by quic-abhijith** [quic/efficient-transformers#147]
**Describe the bug**
QEfficient.cloud.export not using the hf_token passed via cli flag.

**To Reproduce**
1) Make sure no HF_TOKEN flag is set
2) Enable venv, and install efficient-transformer from release 1.18 branch (python 3.8)
3) execute the following command
`QEfficient.cloud.export {}model_name meta-llama/Meta-Llama-3-8B {-}-hf-token <valid_hf_token>`
4) Failure is observed 
The command works, when "export HF_TOKEN=<valid_hf_token>" is provided

**Expected behavior**
Command completes execution and model is exported

**Environment (please complete the following information):**
 - OS: ubuntu 22.04
 - Version/Branch/Commit ID - release/v1.18

Error - 

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "<path_to_efficient_transformers>/QEfficient/cloud/export.py", line 123, in <module>
    main(**args.__dict__)
  File "<path_to_efficient_transformers>/QEfficient/cloud/export.py", line 91, in main
    get_onnx_model_path(
  File "<path_to_efficient_transformers>/QEfficient/cloud/export.py", line 51, in get_onnx_model_path
    _, onnx_model_path = qualcomm_efficient_converter(
  File "<path_to_efficient_transformers>/QEfficient/exporter/export_hf_to_cloud_ai_100.py", line 432, in qualcomm_efficient_converter
    else QEFFCommonLoader.from_pretrained(
  File "<path_to_efficient_transformers>/QEfficient/base/common.py", line 84, in from_pretrained
    pretrained_model_name_or_path = login_and_download_hf_lm(pretrained_model_name_or_path, *args, **kwargs)
  File "<path_to_efficient_transformers>/QEfficient/utils/_utils.py", line 26, in login_and_download_hf_lm
    model_path = hf_download(
  File "<path_to_efficient_transformers>/QEfficient/utils/_utils.py", line 70, in hf_download
    return model_path
UnboundLocalError: local variable 'model_path' referenced before assignment 
```

**Comment by ochougul** [comment]
It's getting fixed in #145 , you can checkout to the PR and test.
This will be merged soon.

**Comment by ochougul** [comment]
We have merged the PR, let us know if the issue persists!
